app/Jobs/CombineExportedPostBatchesJob.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Ushahidi\App\Jobs;
+
+use Exception;
+use RuntimeException;
+use Log;
+use Ushahidi\Core\Entity\ExportJobRepository;
+use Ushahidi\Core\Entity\ExportBatchRepository;
+use Illuminate\Http\File;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Str;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\File as LocalFilesystem;
+
+class CombineExportedPostBatchesJob extends Job
+{
+    protected $jobId;
+
+    protected $csvPrefix = 'csv/tmp';
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($jobId)
+    {
+        $this->jobId = $jobId;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(
+        ExportJobRepository $exportJobRepo,
+        ExportBatchRepository $exportBatchRepo
+    ) {
+        // Load job
+        $job = $exportJobRepo->get($this->jobId);
+
+        if ($job->status === 'completed') {
+            Log::debug('Job already completed', ['jobId' => $this->jobId]);
+            // All done here
+            return;
+        }
+
+        // Check if all batches exported
+        if (!$exportJobRepo->isJobFinished($this->jobId)) {
+            Log::debug('Batches not finished, requeueing', ['jobId' => $this->jobId]);
+            // Batches aren't done yet, wait 5 minutes and try again
+            dispatch(
+                (new CombineExportedPostBatchesJob($this->jobId))
+                    ->delay(Carbon::now()->addMinutes(5))
+            );
+            // All done
+            return;
+        }
+
+        // Load batches
+        $batches = $exportBatchRepo->getByJobId($this->jobId);
+        // Get just filenames
+        $fileNames = $batches
+            ->sortBy('batch_number')
+            ->sortByDesc('has_header')
+            ->pluck('filename')
+            ->all();
+
+        if (!$this->doAllFilesExist($fileNames)) {
+            Log::warning('Some files in export job do not exist', ['jobId' => $this->jobId]);
+            throw new RuntimeException('Some files in export job do not exist');
+        }
+
+        // Combine the files
+        $destinationFile = $this->combineFiles($fileNames);
+
+        // Clean up partial files
+        Storage::delete($fileNames);
+
+        // Set status = completed
+        $job->setState([
+            'url' => Storage::url($destinationFile),
+            'url_expiration' => null, // $urlExpiration, // @todo fix me
```

Comment:
No idea how expirations worked. It looked like we were setting CDN expiries at all!?

Reply:
yea, we were. It's in the ExportBroker.php in the lumen_csvexport_queues repo 
$url_expiration = time() + getenv('TEMP_URL_LIFESPAN');

Reply:
I finally found it in the old code it calls something like `fs->getTempUrl()` to get an expiring URL from RS. I need to figure out how to do that. And how to make sure the bucket isn't public to start with

Reply:
I'm happy to review that with you. From their docs, I don't think there was a way to lock down that bucket otherwise. Any chance we're changing providers?

Reply:
And is this code exclusively for multisite?

Reply:
OSS right now (Jobs won't work with multisite). 
I'm not changing providers right now... but it should work with either of them